TypeSystem/Reflection/types.h

```diff
@@ -62,7 +79,7 @@ struct ReflectedTypeImpl {
 
 #define CURRENT_DECLARE_PRIMITIVE_TYPE(unused_typeid_index, cpp_type, current_type) \
   struct ReflectedType_##current_type : ReflectedTypeImpl {                         \
-    TypeID type_id = TypeID::current_type;                                          \
+    ReflectedType_##current_type() { type_id = TypeID::current_type; }              \
```

Comment:
If `ReflectedTypeImpl` is a pure base class, you could `delete` its default constructor, and only leave the one initializing `type_id`. Its `type_id` would then be `const`.

This way the derived classes would have to use initializer lists.